[Feat/303] 커뮤니티 탭 화면 UI 구현

nahy-512 opened the pull request:
## Related Issue
- #303 

## Type of change
<!-- 작업의 종류를 선택해주세요. -->
- [x] Feature : 새로운 기능 추가
- [ ] Bug fix : 버그 수정
- [ ] Refactor : 코드 리팩토링 작업
- [ ] Document : 문서작업
- [ ] Test : 테스트 코드 작성 및 테스트 작업
- [ ] Style : 코드 스타일 및 포맷팅 작업
- [ ] Chore : 패키지 매니저, 라이브러리 업데이트 등의 작업

## PR Desciption
> 변경 사항 설명

- 커뮤니티 탭의 화면을 구현했습니다.

## Requirements for Reviewer
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

- DataClass는 여기저기에 많이 쓰이는 곳이 있을 거 같아서 안 건드렸습니다. remote, domain 쪽은 API 연동 작업 하면서 정리할 것 같습니다!
- 다이얼로그가 갑자기 이상해졌던데(그래서 구현한 화면들이 디자인과 조금 다르게 나옴), 이건 앱 전체적인 문제 같아서 다른 이슈를 파서 작업하도록 하겠습니다.

## PR Log

> PR 작업하면서 고민했던 내용, 해결한 내용, 고민 중인 내용 등

### 새롭게 배운 것

- AppCompactTextView에 drawableStart, drawableEnd로 넣는 이미지 사이즈를 조절하는 ImageTextView 커스텀뷰를 구현함

### 고민 중인 사항

- 

## 첨부 자료

- [헤이딜러: drawableStart의 이미지 크기를 마음대로 변경하는 방법](https://medium.com/prnd/drawablestart%EC%9D%98-%EC%9D%B4%EB%AF%B8%EC%A7%80-%ED%81%AC%EA%B8%B0%EB%A5%BC-%EB%A7%88%EC%9D%8C%EB%8C%80%EB%A1%9C-%EB%B3%80%EA%B2%BD%ED%95%98%EB%8A%94-%EB%B0%A9%EB%B2%95-950601d6b7db)

nahy-512 committed:
✨ Feat: 커뮤니티 화면 구현

Related to: #303

nahy-512 committed:
✨ Feat: 친구 리스트 화면 검색 필터 추가

Related to: #303

nahy-512 committed:
Merge remote-tracking branch 'origin/dev' into feat/second-tab-303

# Conflicts:
#	app/src/main/java/com/mongmong/namo/presentation/config/RemoteConfigWrapper.kt
#	app/src/main/java/com/mongmong/namo/presentation/utils/BindingAdapters.kt

nahy-512 committed:
✨ Feat: 알림 요청 화면 구현

Related to: #303

nahy-512 committed:
✨ Feat: 모임 알정 생성 화면 구현

- MoimFragment에 플로팅 버튼 추가

Related to: #303

nahy-512 committed:
✨ Feat: 모임 일정 친구 초대 화면 UI 구현

Related to: #303

nahy-512 committed:
✨ Feat: 참석자 캘린더 설정 및 데이터 변경

- 기존 그룹 데이터를 모임 데이터로 이동
- 모임 일정 편집 모드 구현

Related to: #303

nahy-512 committed:
✨ Feat: 모임 참석자 정보 다이얼로그 UI 구현

Related to: #303

nahy-512 committed:
✨ Feat: 게스트 초대 다이얼로그 UI 구현

Related to: #303

nahy-512 committed:
✨ Feat: 친구 정보 다이얼로그 UI 구현

- drawableStart, drawableEnd 이미지 사이즈를 조절할 수 있는 ImageTextView 구현
- 친구 리스트 or 친구 요청에서 진입했는지에 따른 구분

Related to: #303

nahy-512 committed:
✨ Feat: 친구 추가 다이얼로그 UI 구현

Related to: #303

nahy-512 committed:
♻️ Refactor: 친구 캘린더, 모임 캘린더 통합

- CommunityCalendar로 하나로 통합
= isFriendCalendar를 통해 분기 처리

Related to: #303

nahy-512 committed:
♻️ Refactor: v1의 그룹 코드 정리

Related to: #303

nahy-512 committed:
🎨 Style: 모임 일정 참석자 리사이클러뷰 LayoutManager 변경

- Grid -> Flexbox

Related to: #303

nahy-512 committed:
♻️ Refactor: CommunityVPAdapter 위치 변경

Related to: #303

nahy-512 committed:
🐛 Fix: 다이얼로그 패딩 오류 수정

- themes.xml 파일 수정

Related to: #307

edv-Shin commented on app/src/main/res/layout/fragment_moim.xml:
지난 모임일정 숨기기 라디오 버튼의 터치 버블효과가 튑니다

edv-Shin commented on app/src/main/res/layout/fragment_community_calendar_month.xml:
<img width="270" alt="image" src="https://github.com/user-attachments/assets/00948b96-a4c4-4c91-9831-81bd159f37b3">
친구 캘린더, 모임 캘린더에는 뒤로가기 버튼이 있어야 합니다!

edv-Shin commented on app/src/main/java/com/mongmong/namo/presentation/ui/community/CommunityCalendarActivity.kt:
```diff
@@ -0,0 +1,92 @@
+        initClickListeners()
+    }
+
+    private fun setInit() {
```
이미 setup 함수가 있어서 setInit 함수를 따로 빼는것이 불필요해 보입니다.
initClickListeners처럼 기능 or 뷰 단위로 나눠보는게 더 좋을 것 같습니다(ex. setVp..?)

edv-Shin commented on app/src/main/java/com/mongmong/namo/presentation/ui/community/alert/FriendAlertFragment.kt:
```diff
@@ -0,0 +1,56 @@
+    override fun setup() {
+        binding.apply {
+            viewModel = this@FriendAlertFragment.viewModel
+            lifecycleOwner = this@FriendAlertFragment
```
BaseFragment에서 lifecycleOwner 초기화를 이미 해주고 있습니다

nahy-512 committed:
🎨 Design: 커뮤니티 탭 디자인 변경사항 반영

- 커뮤니티 캘린더 뒤로가기 버튼 추가
- 체크박스 리플 효과 삭제
- 모임 일정 하단 생성 버튼 삭제
- 친구 닉네임과 태그 입력란 병합
- 친구 정보 화면 레이아웃 배치 수정

Related to: #303

nahy-512 committed:
🎨 Design: DialogFragment 배경 둥글게 설정

Related to: #303

nahy-512 committed:
🐛 Fix: 커뮤니티 캘린더 피커 선택 후 상단 날짜가 바뀌지 않던 오류 수정

Related to: #303

nahy-512 committed:
♻️ Refactor: 불필요한 코드 정리

- setInit -> initViews
- lifecycleOwner 초기화 코드 정리 (Base와 중복)

Related to: #307
